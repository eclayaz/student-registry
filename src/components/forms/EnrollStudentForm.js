import React from "react";
import { Button, Form, Modal } from "react-bootstrap";

function EnrollStudentForm({
  subjects,
  handleFormSubmit,
  subjectCheckHandler,
  handleEnrollClose,
}) {
  return (
    <form onSubmit={handleFormSubmit}>
      {subjects.map((subject) => (
        <Form.Check
          inline
          label={subject.name}
          type="checkbox"
          key={subject.name}
          name={subject.name}
          checked={subject.value}
          onChange={subjectCheckHandler}
        />
      ))}
      <br />
      <br />
      <Modal.Footer>
        <Button variant="secondary" onClick={handleEnrollClose}>
          Close
        </Button>

        <Button variant="primary" type="submit">
          Add
        </Button>
      </Modal.Footer>
    </form>
  );
}

export default EnrollStudentForm;
