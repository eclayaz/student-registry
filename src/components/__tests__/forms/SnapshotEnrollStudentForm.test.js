import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import EnrollStudentForm from "../../forms/EnrollStudentForm";

const subjects = [
  { name: "English", value: true },
  { name: "Maths", value: false },
];

describe("<EnrollStudentForm /> shallow check", () => {
  it("renders with subject data", () => {
    let wrapped = shallow(<EnrollStudentForm subjects={subjects} />);
    expect(toJson(wrapped)).toMatchSnapshot();
  });
});
