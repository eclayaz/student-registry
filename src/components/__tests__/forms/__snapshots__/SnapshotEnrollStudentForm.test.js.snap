// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnrollStudentForm /> shallow check renders with subject data 1`] = `
<form>
  <FormCheck
    checked={true}
    disabled={false}
    inline={true}
    isInvalid={false}
    isValid={false}
    key="English"
    label="English"
    name="English"
    title=""
    type="checkbox"
  />
  <FormCheck
    checked={false}
    disabled={false}
    inline={true}
    isInvalid={false}
    isValid={false}
    key="Maths"
    label="Maths"
    name="Maths"
    title=""
    type="checkbox"
  />
  <br />
  <br />
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      type="button"
      variant="secondary"
    >
      Close
    </Button>
    <Button
      active={false}
      disabled={false}
      type="submit"
      variant="primary"
    >
      Add
    </Button>
  </ModalFooter>
</form>
`;
